src/node_file.cc

```diff
@@ -1079,41 +1081,101 @@ static void InternalModuleReadJSON(const FunctionCallbackInfo<Value>& args) {
   }
 
   const size_t size = offset - start;
-  char* p = &chars[start];
-  char* pe = &chars[size];
-  char* pos[2];
-  char** ppos = &pos[0];
-
-  while (p < pe) {
-    char c = *p++;
-    if (c == '\\' && p < pe && *p == '"') p++;
-    if (c != '"') continue;
-    *ppos++ = p;
-    if (ppos < &pos[2]) continue;
-    ppos = &pos[0];
-
-    char* s = &pos[0][0];
-    char* se = &pos[1][-1];  // Exclude quote.
-    size_t n = se - s;
-
-    if (n == 4) {
-      if (0 == memcmp(s, "main", 4)) break;
-      if (0 == memcmp(s, "name", 4)) break;
-      if (0 == memcmp(s, "type", 4)) break;
-    } else if (n == 7) {
-      if (0 == memcmp(s, "exports", 7)) break;
-      if (0 == memcmp(s, "imports", 7)) break;
+  simdjson::ondemand::parser parser;
+  auto document = parser.iterate(&chars[start], size, sizeof(&chars[start]));
+  simdjson::ondemand::object obj;
+
+  if (document.get_object().get(obj)) {
+    args.GetReturnValue().Set(Array::New(isolate));
+    return;
+  }
+
+  auto js_string = [&](std::string_view sv) {
+    return ToV8Value(env->context(), sv, isolate).ToLocalChecked();
+  };
+
+  bool includes_keys{false};
+  Local<Value> name = Undefined(isolate);
+  Local<Value> main = Undefined(isolate);
+  Local<Value> exports = Undefined(isolate);
+  Local<Value> imports = Undefined(isolate);
+  Local<Value> type = Undefined(isolate);
+  bool should_parse_exports{false};
+  bool should_parse_imports{false};
+
+  // Check for "name" field
+  std::string_view name_value{};
+  if (!obj["name"].get_string().get(name_value)) {
+    name = js_string(name_value);
+    includes_keys = true;
+  }
+
+  // Check for "main" field
+  std::string_view main_value{};
+  if (!obj["main"].get_string().get(main_value)) {
+    main = js_string(main_value);
+    includes_keys = true;
+  }
+
+  // Check for "exports" field
```

Comment:
You are repeatedly calling `obj["exports"]` which is not ideal. Try...

```
  simdjson::ondemand::value val;
  if(! obj["exports"].get(val)) { // if getting key "exports" worked, then the result is in 'val'
    simdjson::ondemand::json_type exports_type = val.type();
    if(exports_type == simdjson::ondemand::json_type::string) {
        if (!value.get_string().get(exports_value)) {
         exports = js_string(exports_value);
         includes_keys = true;
       } else if (exports_type == simdjson::ondemand::json_type::object) {
       simdjson::ondemand::object subobject;
       if (!value.get_object().get(subobject)) {
         if (!subobject.raw_json().get(exports_value)) {
           exports = js_string(exports_value);
           should_parse_exports = true;
           includes_keys = true;
         }
       }
    }
     
    
  }
```

Or something like it...

Reply:
If you are going to handle exceptions, you can make it shorter...

```C++
  simdjson::ondemand::value val;
  if(! obj["exports"].get(val)) { // if getting key "exports" worked, then the result is in 'val'
    simdjson::ondemand::json_type exports_type = val.type();
    if(exports_type == simdjson::ondemand::json_type::string) {
         exports = js_string(value.get_string());
         includes_keys = true;
     } else if (exports_type == simdjson::ondemand::json_type::object) {
           exports = js_string(value.get_object().raw_json());
           should_parse_exports = true;
           includes_keys = true;
       }
    }
  }
```

Reply:
`simdjson::ondemand::json_type exports_type = val.type();` this line throws error.

```
error: no viable conversion from 'simdjson_result<simdjson::arm64::ondemand::json_type>' to 'simdjson::ondemand::json_type'
```

Reply:
Yes. Bad type. You are right.

Reply:
You can just do `auto exports_type = val.type();`